src/language/printer.ts

```diff
@@ -55,9 +56,47 @@ const printDocASTReducer: ASTReducer<string> = {
   SelectionSet: { leave: ({ selections }) => block(selections) },
 
   Field: {
-    leave({ alias, name, arguments: args, directives, selectionSet }) {
+    leave({
+      alias,
+      name,
+      arguments: args,
+      directives,
+      selectionSet,
+      required,
+    }) {
+      const buildRequiredDesignator = (
+        requiredStatus: ComplexRequiredStatus,
+        accumulator: string = '',
+        depth: number = 0,
+      ): string => {
+        let tempAccumulator = accumulator;
+        if (requiredStatus.subStatus) {
+          tempAccumulator = buildRequiredDesignator(
+            requiredStatus.subStatus,
+            tempAccumulator + '[',
+            depth + 1,
+          );
+        }
+
+        if (requiredStatus.status === RequiredStatus.REQUIRED) {
+          tempAccumulator += '!';
+        } else if (requiredStatus.status === RequiredStatus.OPTIONAL) {
+          tempAccumulator += '?';
+        }
+        if (depth > 0) {
+          tempAccumulator += ']';
+        }
+
+        return tempAccumulator;
+      };
+
       const prefix = wrap('', alias, ': ') + name;
-      let argsLine = prefix + wrap('(', join(args, ', '), ')');
+      const requiredDesignator = buildRequiredDesignator(
+        required as unknown as ComplexRequiredStatus,
```

Comment:
Typescript thinks that `required` is a `string`, but it's actually a `ComplexRequiredStatus`. I'm not sure why that's happening, so this is how I'm dealing with it. Let me know if there's something else I can do.